lib/grape/dsl/request_response.rb

```diff
@@ -104,7 +104,7 @@ def rescue_from(*args, &block)
             handler = block
           end
 
-          options = args.extract_options!
+          options = args.extract_options!.dup
```

Comment:
> This particular one actually returns a new hash every time

Oh, shame to me =)

Ok, we should have discussed it a little bit before I made my changes =). The option ```:with``` was used in middleware/error. I removed that usage, and stored its value in ```rescue_handlers``` hash. 

So I decided to delete ```:with``` from ```options``` hash just to clean it from unused key. And that's why I had to fix request_response_spec.rb.